### PR TITLE
Future RFC - Asynchronously import hidden components

### PR DESCRIPTION
**Pull request by czearing**
This RFC proposes the idea of asynchronously importing hidden components such as `Tooltip` and `ContextualMenu` to assist in bundle size related issues.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr>                                           |   `55.21 kB`<br />`17.401 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |   `56.558 kB`<br />`15.66 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                                                                       |   `24.343 kB`<br />`7.165 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                                                                         |   `27.156 kB`<br />`7.851 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenseBadge</abbr>                                                                       |          `237 B`<br />`177 B` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                                                                    |     `25.03 kB`<br />`8.04 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                                                                    |   `30.322 kB`<br />`8.917 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                                                            |   `26.617 kB`<br />`8.549 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                                                                        |   `34.627 kB`<br />`8.676 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `166.545 kB`<br />`48.067 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `36.258 kB`<br />`11.596 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                                                                 |   `15.889 kB`<br />`5.747 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                                                                       |   `10.642 kB`<br />`4.264 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                                                                       |    `9.397 kB`<br />`3.839 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                                                                         |   `14.715 kB`<br />`6.012 kB` |
| <samp>react-make-styles</samp> <br /> <abbr title='bundle-size/makeStaticStyles-runtime.fixture.js'>makeStaticStyles (runtime)</abbr>                                         |     `7.59 kB`<br />`3.321 kB` |
| <samp>react-make-styles</samp> <br /> <abbr title='bundle-size/makeStyles-runtime.fixture.js'>makeStyles + mergeClasses (runtime)</abbr>                                      |   `22.135 kB`<br />`8.356 kB` |
| <samp>react-make-styles</samp> <br /> <abbr title='bundle-size/makeStyles.fixture.js'>makeStyles + mergeClasses (build time)</abbr>                                           |    `2.557 kB`<br />`1.202 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          | `106.042 kB`<br />`32.735 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `108.802 kB`<br />`33.277 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `102.804 kB`<br />`31.283 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |     `7.78 kB`<br />`2.672 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePopper</abbr>                                                                         |   `23.145 kB`<br />`7.942 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                                                                  |   `16.256 kB`<br />`5.969 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |   `28.975 kB`<br />`9.115 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                                                                        |   `11.798 kB`<br />`4.452 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                                                                      |   `15.414 kB`<br />`4.734 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsAllThemes.fixture.js'>Teams: all themes</abbr>                                                                  |   `32.941 kB`<br />`6.674 kB` |
| <samp>react-theme</samp> <br /> <abbr title='bundle-size/TeamsLightTheme.fixture.js'>Teams: Light theme</abbr>                                                                |   `20.247 kB`<br />`5.662 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `46.054 kB`<br />`15.658 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                                                                       |          `213 B`<br />`170 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e429b5325e128d7cc6068a208c5f9e87b9ce09ca'>e429b5325e128d7cc6068a208c5f9e87b9ce09ca</a></sub>
<div id="bundleSizeComment9423"/>

File: rfcs/convergence/async-import-hidden-components.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Components such as `Tooltip` are split into 2 parts:
+
+1. A synchronous tiny component
+2. An asynchronous heavy component.
+
+When a consumer imports `Tooltip`, they get the tiny part in their bundle which just registers handlers and logic to determine when the tooltip is visible.
+
+Once it becomes visible, we async import the actual `Tooltip` code, which in turn has graph edges to heavy things like `Popper` and styling.
+
+We should consider async imports usage for things not initial-render-critical in converged components. All major bundlers support async imports. For scenarios where it does not such as AMD bundlers and old builds of webpack, partners are free to use a pre-bundle step to produce an AMD friendly bundle using a modern bundler. Alternatively use Fluent's UMD bundle.
```
Comment (ecraig12345) [inline rfcs/convergence/async-import-hidden-components.md:40]: > Alternatively use Fluent's UMD bundle.

One small thing, we don't currently have a UMD bundle, though we could potentially add one in the future if it's needed. (Background: We have a UMD bundle in v8 to support codepen and maybe basic prototyping scenarios, but nobody is going to use a UMD bundle in a real app because it's huge.)

File: rfcs/convergence/async-import-hidden-components.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+## Detailed Design or Proposal
+
+Components such as `Tooltip` are split into 2 parts:
+
+1. A synchronous tiny component
+2. An asynchronous heavy component.
+
+When a consumer imports `Tooltip`, they get the tiny part in their bundle which just registers handlers and logic to determine when the tooltip is visible.
+
+Once it becomes visible, we async import the actual `Tooltip` code, which in turn has graph edges to heavy things like `Popper` and styling.
```
Comment (ecraig12345) [inline rfcs/convergence/async-import-hidden-components.md:38]: It might be helpful to have some code snippets of what this would look like. The import syntax is pretty standard but it's good to have it explicitly shown to ensure everybody is on the same page.

File: rfcs/convergence/async-import-hidden-components.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+Currently, we are avoiding using our own components in meaningful places, with (unnecessary) concern of bundle bloat due to this issue.
+
+Partners will likely do the same. (e.g. try to use `Tooltip`, find out they regressed 10k on initial paint impact and use an alternative solution.)
+
+## Detailed Design or Proposal
+
+Components such as `Tooltip` are split into 2 parts:
+
+1. A synchronous tiny component
```
Comment (ecraig12345) [inline rfcs/convergence/async-import-hidden-components.md:33]: Can we get info about the actual size of the "tiny" part for something like Tooltip or Menu?

**Review by ecraig12345 — COMMENTED** [review]
Thanks for driving this @czearing, and thanks @dzearing for providing the background! I think this makes sense but I'm curious to get input from people on @microsoft/teams-prg and/or anyone else who might be more familiar with partner scenarios to ensure we're not missing any potential issues.

File: rfcs/convergence/async-import-hidden-components.md (hunk 3 above)
Comment (czearing) [reply rfcs/convergence/async-import-hidden-components.md:33]: I just added a snippet of the logic from the `Tooltip` render to showcase the size.

File: rfcs/convergence/async-import-hidden-components.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+When a consumer imports `Tooltip`, they get the tiny part in their bundle which just registers handlers and logic to determine when the tooltip is visible.
+
+```jsx
+<>
+{state.children}
+{state.shouldRenderTooltip && (...)}
```
Comment (jurokapsiar) [inline rfcs/convergence/async-import-hidden-components.md:40]: can you please add to the example how we would apply aria-label / labelledby while the full tooltip component is not loaded?

Comment (ecraig12345) [reply rfcs/convergence/async-import-hidden-components.md:40]: I'm not sure it would ever work properly to use a tooltip as the target of aria-labelledby, regardless of whether it's synchronously or async loaded (unless I'm misunderstanding what you mean). @smhigley could confirm.

File: rfcs/convergence/async-import-hidden-components.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+### Pros and Cons
+
+#### Pros
+
+- When partners use the `Tooltip`, their main bundle size stays small, and the page load time hit is minimized by default.
+- Components such as `Slider` can use the `Tooltip` directly without worrying as much about bundle impact.
+
+#### Cons
+
+- Partners who use old bundlers would potentially be affected.
```
Comment (ling1726) [inline rfcs/convergence/async-import-hidden-components.md:87]: If a user does not use something like create-react-app and uses a minimal webpack config, would that break their app until they add the approriate code splitting plugin ?

Comment (ecraig12345) [reply rfcs/convergence/async-import-hidden-components.md:87]: Webpack should handle that automatically (no plugins needed) since `import()` is standard syntax now.

Comment (ecraig12345) [reply rfcs/convergence/async-import-hidden-components.md:87]: This is the semi-relevant section in the docs. It's not super explicit about this but from what's there and my experience I'm still pretty sure it will "just work." SplitChunksPlugin or whatever is only needed if you want to have highly customized splitting bheavior. https://webpack.js.org/guides/code-splitting/#dynamic-imports

**Review by khmakoto — COMMENTED** [review]
I don't know if this is an issue or not, but are there any concerns in using async imports on SSR scenarios?

**Comment by czearing** [comment]
> I don't know if this is an issue or not, but are there any concerns in using async imports on SSR scenarios?

Good question. Async imports work as long as you are using node 14 with modules however `React.lazy` is yet to support SSR. There are other code splitting tools such as Loadable Components (this is what React suggests using for SSR)  that could support it. However I believe that some hidden components such as Tooltip would not function correctly in SSR since they require positioning logic and hover events to become visible.

File: rfcs/convergence/async-import-hidden-components.md (hunk 4 above)
Comment (jurokapsiar) [reply rfcs/convergence/async-import-hidden-components.md:40]: I mean the fucntionality of applying label/description based on `triggerAriaAttribute` prop on the trigger: https://github.com/microsoft/fluentui/blob/789a3733b128569190319fce3fe2b46900b24896/packages/react-tooltip/src/components/Tooltip/useTooltip.tsx#L218

how would that work when there is only the minimal part loaded?

File: rfcs/convergence/async-import-hidden-components.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+```jsx
+const RenderTooltipContent = React.lazy(() => import('./renderTooltipContent'));
+
+export const renderTooltip = (state: TooltipState) => {
+  return (
+    <>
+      {state.children}
+      {state.shouldRenderTooltip && (
+        <React.Suspense fallback={null}>
```
Comment (layershifter) [inline rfcs/convergence/async-import-hidden-components.md:67]: - What happens when this fails to load/loads after `.shouldRenderTooltip` becomes `true`? Customers will see nothing?
- SSR is not supported by `.Suspense` ➡ we can't rely on it currently

----

Please check for inspiration: https://github.com/theKashey/use-sidecar
I haven't checked how it works, but it's very close to our goal I guess

File: rfcs/convergence/async-import-hidden-components.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+In the past we've avoided the use of async imports to prescribe split points, because the technology was new, and bundlers didn't have strong support for it. We built a proprietary webpack plugin for managing re-aliasing. This was done before webpack supported async import split points, so that we could ensure the right files were aliased, rather than asking each partner to set this up. But (I believe) the only partner to take advantage of this plugin was OOUI. (Was very proprietary, not well documented, etc.) But now with async imports, we don't need plugins. With the new components being worked on and tools being modernized, we should revisit this issue.
+
+If you don't know what they are; it is a way to asynchronously import source code. Bundlers use async imports as a cue for creating a split point in the bundle.
+
+Read more about how React itself helps make this simple for components: https://reactjs.org/docs/code-splitting.html
+
+## Problem statement
+
+We have several scenarios where async imports would be beneficial for minimizing bundle size impact on our partners apps, especially for initial page load.
```
Comment (GeoffCoxMSFT) [inline rfcs/convergence/async-import-hidden-components.md:21]: Users with limited bandwidth or where the last mile is distant from origin, the cost of each JS download can be mostly the request resolution and server connection.  Once the data starts downloading, the payload size doesn't cost as much relative to the connection.  In other words, if there are lots of small requests it can be more expensive.  

This is typically solved with CDNs or edge caching. 
- What the experience would be with devs wanting to use a CDN? 
- Would each dynamic import be able to be resolved to a CDN address?  
- Does having lots of code splits mean that there are more things being invalidated as fluent changes?

File: rfcs/convergence/async-import-hidden-components.md (hunk 3 above)
Comment (GeoffCoxMSFT) [reply rfcs/convergence/async-import-hidden-components.md:33]: Would dynamic imports might be most useful in the icon case to only bring in the component with the SVG when used?

File: rfcs/convergence/async-import-hidden-components.md (hunk 7 above)
Comment (ecraig12345) [reply rfcs/convergence/async-import-hidden-components.md:21]: [SplitChunksPlugin](https://webpack.js.org/plugins/split-chunks-plugin/) has options to help configure this on the consumer side, including without any knowledge of paths/filenames. Most nontrivial apps will already have async imports and multiple bundles, and probably have SplitChunksPlugin or a similar solution set up to customize it. So I don't think this is something we need to worry about directly.

**Comment by ecraig12345** [comment]
Meeting notes:
- Need to look at specific major partners, how they're handling bundling/splitting, what their priorities are/why they're splitting, and how they could customize if we're splitting
- If we don't use async imports, how do we handle the problem that motivated this proposal? (want to have a tooltip or some other on-demand feature in a component without it getting included in PLT bundle)
- Will this create issues for using apps offline? (pretty sure this is an app concern and something they'd configure, not that we'd configure)
- Can we have a way to support both sync/async?
  - Ideally we would be able to support both options a Tooltip component as well as an async version such as AsyncTooltip. This way an end user won't be surprised when a presentational component suddenly behaves asyncronously behind the scenes but at the same time they have a way to opt-into this async behaviour provided by React natively without the need to code it themselves.
  - webpack or babel plugin for setting specific components to lazy? (might need to have goal of low config)
- How to address SSR issue already commented on?
  - There is some work on SSR in React 18: [New Suspense SSR Architecture in React 18](https://github.com/reactwg/react-18/discussions/37)

**Comment by ecraig12345** [comment]
Discussed this again and the conclusion was that we should close it for now (and make a follow-up issue) since a lot more investigation is needed and it's not immediately blocking any critical scenarios. Thanks @czearing and @dzearing for starting the discussion though!

File: rfcs/convergence/async-import-hidden-components.md (hunk 3 above)
Comment (dzearing) [reply rfcs/convergence/async-import-hidden-components.md:33]: tiny part is the trigger code required to "start" showing the menu/tooltip. likely less< 1k.

File: rfcs/convergence/async-import-hidden-components.md (hunk 6 above)
Comment (dzearing) [reply rfcs/convergence/async-import-hidden-components.md:67]: If a javascript file fails to load on web app startup or on an async on demand bundle, this is essentially out of scope of fluent ui. Some webpack partners override the webpack async loading logic to have retries and fallbacks and ultimately a BSOD equivallent.

**Comment by dzearing** [comment]
I think closing this prematurely is a bad idea. Essentially, it changes the design of a component significantly. You need to separate your logic into the "synchronous" inexpensive part from the "async" expensive part.

It doesn't have to be complicated, and it doesn't have to apply to every component. But we're literally working on tooltip and menu now, so really this should be considered as part of the design in those components.

But history has shown here that ignoring bundle size leads to customers being frustrated, and we've done this sort of thing in the past. I think in the past there weren't standard practices involved here. But there are now. So we should design for them.

IRT meeting notes:

> Need to look at specific major partners, how they're handling bundling/splitting, what their priorities are/why they're splitting, and how they could customize if we're splitting

SHysom is a great resource here. But I can already tell you this - this RFC originated from conversations with him and others on his team working on Ribbon perf. It is really important that we avoid bundle penalties for using Tooltips and Menus. They are used frequently in Ribbon and other Office scenarios.

A couple years ago, one dev was evaluating Menus and worked closely with KChau on how to defer the menu loading code because it brought in a swarm of dependencies. It was painful because we didn't have tools and Ken wrote a webpack plugin specifically to support this scenario because there wasn't a great way on the app consumer side to know exactly where/how to split. It wasn't considered in the design.

We have an opportunity with converged components to design these limited scenarios correctly so that bundle size isn't penalized. Ignoring it will likely repeat history.

I would really urge re-opening the RFC and requiring the Menu/Tooltip devs hash out what's required to avoid penalizing devs initial app downloads by 30k. Limit it to those 2 controls, create a plan to try designing for this, without things being set in stone. Then once the pattern is solidified and the issues called out in the RFC (such as SSR) have been addressed/avoided, update the RFC and finish merging this.

**Comment by ecraig12345** [comment]
@dzearing Ah okay, I wasn't aware that this was immediately impactful for Menu and Tooltip (I thought the motivating scenario was specifically the tooltip on Slider, which is definitely less urgent). cc @ling1726 and @behowell 

The major blocker here is lack of support for async imports in Node without newer versions and/or experimental flags, causing issues in SSR and maybe tests (though tests have to use commonjs still for related reasons). Lack of support for React lazy/suspense in SSR is also an issue. Unless we can find a way around those things, we probably can't use async imports in our code directly. 

So then the question becomes what we *can* do instead--there were some ideas, but all of them need further investigation.
- For components which we'd want to load async, user must pass them in through slots/props (would have to figure out exact API)
  - This could be hard to mesh with our existing slot API: in some cases, to avoid requiring consumers to have knowledge of exactly what a given component's "built-in" tooltip ought to look like/contain, they'd need to pass the component itself not a rendered version
- Use a provider/context to customize loading
  - potential issue: webpack requires use of string literals in imports for code splitting/tree shaking (can't pass a path name to a custom loader function)
- Some other user-customizable loader for loading things either sync or async as desired (@varholak-peter had an idea about this)

Alternatively, it *might* be possible to say (and document) that SSR is an edge case where it's okay to require extra config, and provide or recommend a babel plugin or something to convert all async imports to sync. This would probably be okay with our major partner teams (since AFAIK none of them use SSR?) but seems a bit dismissive of the larger community. 

(Another quick note on conversion plugins: writing all imports as sync and using a plugin to selectively convert to async is much more complicated and may not be feasible.)

Even if we find a way around the SSR issues and decide to start using async imports in our code, there are still quite a few outstanding questions about the implications in different scenarios:
- What code splitting approach are partners using today (automatic, chunk count or size, specific filenames/entry points, other?) and would introducing async imports into our own code create issues for that?
- Would this make it harder for partners to make their apps work offline?
  - My guess is no, because a nontrivial app would already be using async imports and multiple bundles internally, so extending that configuration to our code would *probably* not be an issue
  - More generally, it would be good to understand and document what would a partner need to do to make their app work offline (using webpack, other bundlers, and/or any other configuration required)
- Basic create-react-app: does this result in multiplication of tiny chunks, what needs to be done for it to work offline, can either of the above be customized

Basically we need a lot more investigation, including demos of various approaches, and the consensus among those who were in the meeting this morning was that we just don't have time for that right now. However that consensus was not considering the general issues you raised about menu and tooltip triggers--we didn't discuss that at all since we weren't aware that was an immediate concern.

**Comment by dzearing** [comment]
If in your typescript code you say:

```ts
export function getFoo() {
  import("./Foo").then((foo) => {
    console.log(foo);
  });
}
```

...and you compile to commonjs modules, it produces this:

```js
"use strict";
exports.__esModule = true;
exports.getFoo = void 0;
function getFoo() {
    Promise.resolve().then(function () { return require("./Foo"); }).then(function (foo) {
        console.log(foo);
    });
}
exports.getFoo = getFoo;
```

Which I believe is compatible with node 12 and even older (wherever Promises were introduced.)

**Comment by ecraig12345** [comment]
@dzearing Maybe a silly question (I'm not familiar with this area) but is it expected in general that SSR would have to use the commonjs version of the code?

**Comment by dzearing** [comment]
pretty likely; I think when you import a `.js` file, even the latest node versions will assume commonjs unless you do one of the following:

* You rename the extension to `.mjs`
* You provide `"type": "module"` in your `package.json` to indicate that `.js` files should be parsed as ESM.

One thing I'm not totally sure on however is if it infers the module default if the initial resolution was due to `module` entry point. That is, you import `@fluentui/react-tooltip`, which has a `module` entry to `index.js`. That file then exports * from `./Tooltip.js`. Will both files be treated as ESM because `module` was used to resolve the entry? I don't know that one, needs some experimentation. I can try this out and see what happens.

**Comment by dzearing** [comment]
Also, I assume this edge case isn't an issue for older versions that don't support async imports (v12?) because likely they would also not try to load the entry from `module` if ESM isn't enabled, thus falling back to `main` which would point to the commonjs files.

**Comment by levithomason** [comment]
One idea that comes to mind is that this might be a good candidate for a v10 change. My thoughts are that we're mid releasing beta and a stable 9.0.0 soon after and async, while valuable, is likely to be a significant departure from the current architecture.

We have many good things to provide in v9 as it stands and should continue with shipping that, but we should also continue this async proposal. I would second @dzearing's comments about bundle size issues for importing things that are not used on first load. It was one of the things we discussed in the very early days of convergence as both libs suffered from issues with sync interdependencies like this.

All said, this topic deserves a proper amount of time to think through it and test. I don't see that fitting on the 9.0 runway, especially given that we're currently cutting items to make the beta/major targets. Why don't we continue this conversation but give ourselves some breathing room and considering it a v10 proposal?

**Comment by dzearing** [comment]
For RFCs that are planning to be re-visited, is there an rfc folder for "v10" that this can be merged into (where all the v10 plans are), so that there's an easy way to come back to it?

**Comment by ecraig12345** [comment]
We do have a `rfcs/future` folder, so maybe it could go there?

**Comment by czearing** [comment]
I'm reopening the pull request to move the RFC into the future folder so it can be re-visited.

**Review by khmakoto — APPROVED** [review]
Approving now that this is being moved into the future folder, although maybe we should make a note about how we are continuing this conversation later at the beginning or end of the RFC.

**Review by ecraig12345 — APPROVED** [review]
I just pushed an update adding myself to the list of drivers, switching the comparison to gzip sizes (this is preferred because it's what would actually be downloaded by end users' computers), and clarifying which section of the proposal is the original part that hasn't yet been updated from PR discussion. With that it LGTM. Thanks for working on this!
